frontend/javascripts/oxalis/controller/scene_controller.ts

```diff
@@ -634,6 +656,20 @@ class SceneController {
           this.updateUserBoundingBoxesAndMeshesAccordingToTransforms();
         },
       ),
+      listenToStoreProperty(
+        (storeState) => {
+          const visibleSegmentationLayerNames = getVisibleSegmentationLayers(storeState).map(
+            (l) => l.name,
+          );
+          if (_.isEqual(this.visibleSegmentationLayerNames, visibleSegmentationLayerNames)) {
+            return this.visibleSegmentationLayerNames;
+          } else {
+            this.visibleSegmentationLayerNames = visibleSegmentationLayerNames;
+            return visibleSegmentationLayerNames;
+          }
+        },
```

Comment:
You can use `reuseInstanceOnEquality` for this if I'm not mistaken. It would look something like this:

```ts
// top level
const getVisibleSegmentationLayerNames = reuseInstanceOnEquality(
          (state) => getVisibleSegmentationLayers(storeState).map(
            (l) => l.name,
          )
)

// in listenToStoreProperty
return getVisibleSegmentationLayerNames(storeState)
```


the instance variable `this.visibleSegmentationLayerNames` can be removed then, too.

Reply:
Very awesome thanks :tada: